### DISCUSSION TITLE
Failed to send a little bit large file.

### DISCUSSION THREAD

**Discussion by RemonAhammad** [libp2p/rust-libp2p#4438] [Q&A] [answered]
I just copied the full code of your file-sharing example project. I can send a small file like a 1.1 MB file (I have tested). 

but the problem is I can't send a little bit large file around 25 MB.

please go through my [code](https://github.com/blackshadowsoftwareltd/libp2p_file_sharing)

the error occurred : (File Sender terminal)
```
cargo run -- --listen-address /ip4/127.0.0.1/tcp/40837 \
          --secret-key-seed 1 \
          provide \
          --path /home/remon/Desktop/extra_large.jpg \
          --name d_img
   Compiling libp2p_file_sharing v0.1.0 (/home/remon/Rusty/libp2p_file_sharing (copy))
    Finished dev [unoptimized + debuginfo] target(s) in 16.44s
     Running `target/debug/libp2p_file_sharing --listen-address /ip4/127.0.0.1/tcp/40837 --secret-key-seed 1 provide --path /home/remon/Music/video.zip --name d_img`
Command::StartListening
Local node is listening on "/ip4/127.0.0.1/tcp/40837/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X"
Command::StartProviding
StartProviding
IncomingConnection
ConnectionEstablished 12D3KooWBF6zdaLuWsUyVBnrvCpfoCUntQK3j54t9EBuQ1ViXgYT
-------------------
-------------------
Request
Command::RespondFile
ConnectionClosed
thread 'async-std/runtime' panicked at 'Behaviour(RequestResponse(InboundFailure { peer: PeerId("12D3KooWBF6zdaLuWsUyVBnrvCpfoCUntQK3j54t9EBuQ1ViXgYT"), request_id: RequestId(13), error: ConnectionClosed }))', src/network.rs:356:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'not yet implemented: None', src/main.rs:67:26
```
the error occurred : (File Receiver terminal)
```
╰─ cargo run -- --peer /ip4/127.0.0.1/tcp/40837/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X \
          get \
          --name d_img
    Finished dev [unoptimized + debuginfo] target(s) in 0.53s
     Running `target/debug/libp2p_file_sharing --peer /ip4/127.0.0.1/tcp/40837/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X get --name d_img`
Command::StartListening
Command::Dial
-------------------
Local node is listening on "/ip4/127.0.0.1/tcp/43719/p2p/12D3KooWBF6zdaLuWsUyVBnrvCpfoCUntQK3j54t9EBuQ1ViXgYT"
Local node is listening on "/ip4/192.168.1.84/tcp/43719/p2p/12D3KooWBF6zdaLuWsUyVBnrvCpfoCUntQK3j54t9EBuQ1ViXgYT"
ConnectionEstablished 12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X
Command::GetProviders
-------------------
FoundProviders
FinishedWithNoAdditionalRecord
File name : "d_img"
Command::RequestFile
ConnectionClosed
OutboundFailure
Error: "None of the providers returned file."

```

**Comment by thomaseizinger** [comment, marked as answer]
Yes, there is a maximum amount that we will allocate: https://github.com/libp2p/rust-libp2p/blob/955da49f26e9bad979bd24fc008cc9c1675f7c71/protocols/request-response/src/cbor.rs#L55-L58

This MUST be bounded otherwise you are vulnerable to DoS attacks because the remote can send you random garbage and you allocate 10MB per stream and eventually run out of memory. If you want to transfer large files, you'll need to chunk them somehow.

**Reply by thomaseizinger** [reply]
The file sharing example is just an _example_. See https://github.com/libp2p/rust-libp2p/blob/master/examples/file-sharing/README.md#architectural-properties for what it focuses on. Actually building an application that can be used for sharing arbitrarily large files is more involved than this :)

It might be useful to document these limits in the README. Are you open to send a PR for that?

**Reply by RemonAhammad** [reply]
Is there any new update about this discussion?
Can I get an example of sending a large file chunk chunk?

**Reply by thomaseizinger** [reply]
At this point, this is the only example we have. Sending arbtirarily large files is actually a hard problem that we can't just solve as an example.

There is plenty of research out there. Look into how bitswap, bittorrent, etc work! :)
